### PR TITLE
Adds histogram in the filter analysis

### PR DESCRIPTION
**Pull request by javierarce**
This PR adds a histogram in the filter analysis form, so users can have a glimpse of the distribution of the data in the numeric row they select. 

Since the histogram is generated from pg_stats, min and max data is not always available. To solve this problem, a custom class was created to query the table and obtain min & max values from the selected row.

Kudos to @fdansv for the work with the histogram generation.

screen shot 2016-07-08 at 14 18 58
 
CR: @xavijam

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/column-data.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    this._configModel = opts.configModel;
+    this._nodeDefModel = opts.nodeDefModel;
+
+    this._initBinds();
+    this._setColumn();
+  },
+
+  _initBinds: function () {
+    this._querySchemaModel = new QuerySchemaModel(null, {
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/column-data.js]: I'd create this model in the `initiailze` method.

**Comment by saleiva** [comment]
OLE!

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/column-data.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      configModel: this._configModel
+    });
+
+    this._query = this._nodeDefModel.querySchemaModel.get('query');
+    this.listenTo(this._querySchemaModel, 'change:status', this._onChangeStatus);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/column-data.js]: no need to add_related_model...?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-form-model.js [hunk 3]
```diff
@@ -51,10 +59,13 @@ module.exports = BaseAnalysisFormModel.extend({
     });
 
     this.on('change:kind change:column', this._updateSchema, this);
+    this.on('change:histogram_stats', this._setSchema, this);
 
     this.listenTo(this._columnOptions, 'columnsFetched', this._setSchema);
     this.listenTo(this._columnRowData, 'columnsFetched', this._setSchema);
+    this.listenTo(this._columnData, 'columnsFetched', this._storeHistogramStats);
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-form-model.js]: There is no need to add those collections with add_related_model?

**Comment by xavijam** [comment]
Just those questions :)

**Comment by fdansv** [comment]
❤️

File: lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-form-model.js (hunk 3 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-form-model.js]: It's a model, that method is not available, right?

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-content-views/analyses/analysis-form-models/filter-form-model.js]: True!

**Comment by xavijam** [comment]
@javierarce, resolve conflicts and wait for green ball. If on-premises tests fail again, ping @jvillarf.

**Comment by jvillarf** [comment]
Hey, yesterday i had the day off, sorry. I'm currently taking a look to the user_mover specs, trying to see why thy are failing.

**Comment by jvillarf** [comment]
@javierarce  could you please relaunch the tests? user_mover specs shouldn't fail anymore. Thanks!

**Pull request merged**